frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx

```diff
@@ -69,6 +69,9 @@ const EditCamperModal = ({
     formResponses,
   };
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const initialStateVariables = useMemo(() => stateVariables, []);
```

Comment:
the initial state needs to be updated everytime we select a new camper or the afformentioned bug will occurr. I solved it by passing in all the fields as dependencies to the useMemo 
```
  const initialStateVariables = useMemo(() => stateVariables, [
    firstName,
    lastName,
    age,
    allergies,
    specialNeeds,
    formResponses,
  ]);
  ```

Reply:
Basically, the issue is that the initial state doesn't get updated properly between renders. Whichever camper I selected first to edit, that camper's initial fields will be saved in `initialStateVariables` so to make sure it's updated between renders I just passed all the selected camper's fields as dependencies. 

We don't need to use this exact implementation but the idea is we need to ensure `initialStateVariables` between renders